sklearn/feature_selection/tests/test_feature_select.py

```diff
@@ -371,6 +371,46 @@ def test_select_heuristics_regression():
         assert_less(np.sum(support[5:] == 1), 3)
 
 
+def test_select_artificial_data_ch2():
```

Comment:
If the test remains this all-encompassing, can we please rename to "test_boundary_case" and comment that we're always aiming to select 1 feature, and testing the boundary case...